data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/SplitStringProcessor.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.mutatestring;
+
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.processor.Processor;
+
+import java.util.List;
+
+@DataPrepperPlugin(name = "split_string", pluginType = Processor.class, pluginConfigurationType = SplitStringProcessorConfig.class)
+public class SplitStringProcessor extends AbstractStringProcessor<SplitStringProcessorConfig.Entry> {
+
+    private final List<SplitStringProcessorConfig.Entry> entries;
+
+    public SplitStringProcessor(PluginMetrics pluginMetrics, StringProcessorConfig<SplitStringProcessorConfig.Entry> config) {
+        super(pluginMetrics, config);
+        this.entries = config.getIterativeConfig();
+    }
+
+    @Override
+    protected void performKeyAction(Event recordEvent, SplitStringProcessorConfig.Entry entry, String value) {
+
+        String[] splitValue = value.split(entry.getDelimiter());
```

Comment:
Instead of compiling the regex each time this function is called, pre-compile the config entries into a `patternMap` like I did in my PR: https://github.com/opensearch-project/data-prepper/pull/1170/files#diff-b9aa04ba8c402bbf7728c0050292c6b6123d6a4b2a73761b5241ecfc05cdf4e4R25

Reply:
+1 to compiling and validating the regex patterns in the constructor

Reply:
I will update this since it'll improve the performance, thanks for the reference.